python-checks/src/main/java/org/sonar/python/checks/cdk/PublicNetworkAccessToCloudResourcesCheck.java

```diff
@@ -51,10 +54,58 @@ protected void registerFqnConsumer() {
     checkFqn("aws_cdk.aws_rds.DatabaseInstance", PublicNetworkAccessToCloudResourcesCheck::checkDatabaseInstance);
 
     checkFqn("aws_cdk.aws_rds.CfnDBInstance", (subscriptionContext, callExpression) ->
-      getArgument(subscriptionContext, callExpression, PUBLICLY_ACCESSIBLE_ARG_NAME).ifPresent(
-        argument -> argument.addIssueIf(isTrue(), MESSAGE)
-      )
-    );
+      getArgument(subscriptionContext, callExpression, PUBLICLY_ACCESSIBLE_ARG_NAME).ifPresent(argument -> {
+        if (argument.hasExpression(isTrue()) && !isSafeNetwork(subscriptionContext, callExpression)) {
+          argument.addIssue(MESSAGE);
+        }
+      }));
+  }
+
+  private static boolean isSafeNetwork(SubscriptionContext subscriptionContext, CallExpression callExpression) {
```

Comment:
This method is much too big and hard to follow. I would split it into smaller methods that can describe the subtask by method name.